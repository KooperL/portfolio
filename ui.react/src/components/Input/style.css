#Input p {
  width: fit-content;
  flex-wrap: nowrap;
  word-wrap: 0;
  white-space: nowrap;
}

#Input-decorator {
  display: flex;
  justify-content: center;
  align-items: center;
  padding-left: 5px;
  background-color: white;
  color: black;
  /* height: 100%; */
  border-top-left-radius: 10px;
  border-bottom-left-radius: 10px;
}

#Input {
  width: 100%;
  height: 2rem;
  display: flex;
  flex-direction: row;
}

#Input-container {
  width: 100%;

}


#Input-container input {
  width: 100%;
  color: black;
  background-color: white;
  /* border-radius: 10px; */
  border-top-right-radius: 10px;
  border-bottom-right-radius: 10px;
  padding: 5px 10px;
}
