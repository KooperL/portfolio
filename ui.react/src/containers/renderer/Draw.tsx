import { triangle } from "./utils/types";

export function DrawFigure(props: {name: string, h: number | string, w: number | string}) {
  // const displayGCD = gcd(+props.w, +props.h);
  return (
    <figure>
      {/* <svg viewBox={`${0-+props.w/2} ${0-+props.h/2} ${+props.w/2} ${+props.h/2}`} width={props.w} height={props.h}> */}
      <svg viewBox={`0 0 ${+props.w} ${+props.h}`} width={props.w} height={props.h}>
        <use xlinkHref={`#${props.name}`}/>
      </svg>
    </figure>
  );
}

export function drawBlueprint(props: {name: string, tris: triangle[]}) {
  return (
    <svg className="defs-only" xmlns="http://www.w3.org/2000/svg">
      <symbol id={props.name}>
        {props.tris.map((triangle, triangleIndex) => (
          <polygon key={triangleIndex}
            points={`${triangle.p[0].x},${triangle.p[0].y} ${triangle.p[1].x},${triangle.p[1].y} ${triangle.p[2].x},${triangle.p[2].y}`}
            style={{
              fill: triangle['c']['fill'] ?? 'red',
              stroke: triangle['c']['stroke'] ?? 'pink',
              strokeWidth: triangle['c']['strokeWidth'] || 5.0,
            }}
          />
          ))}
      </symbol>
    </svg>
  );
}