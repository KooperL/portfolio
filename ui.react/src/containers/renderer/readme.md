Pressing pause for now.
 TODO:
 - Vector transformation as a method of their class. This will let them keep their colour properties when undergoing a transformation
 - Edge plane clipping for simple raster
 - Try to do something to reduce duplicate code
 - Use ticker as switch to toggle different transformations for grid
 - Do the scale properly -> make it a method and take args
 - colour spectrum