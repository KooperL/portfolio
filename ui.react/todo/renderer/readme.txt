Pressing pause for now.
 TODO:
 - Vector transformation as a method of their class. This will let them keep their colour properties when undergoing a transformation
 - Edge plane clipping for simple raster
 - Try to do something to reduce duplicate code
 - Put engine components behind layers of abstraction to be changed without affecting core engine -> make it a method and take args
 - colour spectrum
 - switch between canvas, web-gl enabled canvas and svg rasterisation

Abandoning, performance is simply not going to be acceptable without spending more time learning optimising strategies. Time better spent implementing algo's with a public library. Shame.
KT to myself:
 - V1 had best performance, but function structure was a mess and wasn't designed to be a general engine
 - V2 Was more well made and class based, not optimal for react but function structure was very ideal. Probably where I should pick up
 - V3 Was V2 crammed into react functional components, eventually started impacting performance.