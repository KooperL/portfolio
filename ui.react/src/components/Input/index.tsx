import React from 'react'
import './style.css'

export function Input(props: {
  value: string
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void
  readOnly?: boolean
  autoComplete?: string
  name?: string
  id?: string
  placeholder?: string
  label?: string
  inputBoxLabel?: string
}) {
  return (
    <div id="Input-container">
      <label htmlFor={props.name}>{props.label}</label>
      <div id="Input">
        <div id="Input-decorator">
          <p>{props.inputBoxLabel}&nbsp;</p>
        </div>
        <input
          type="text"
          readOnly={props?.readOnly}
          autoComplete={props?.autoComplete}
          value={props.value}
          onChange={props.onChange}
          name={props.name}
          id={props.id}
          placeholder={props.placeholder}
          // style={{
          //   ...(
          //     !!props.inputBoxLabel ?
          //     {
          //       borderTopLeftRadius: '0',
          //       borderBottomLeftRadius: '0',
          //     } :
          //     {borderRadius: ''}
          //   )
          // }}
        />
      </div>
    </div>
  )
}
