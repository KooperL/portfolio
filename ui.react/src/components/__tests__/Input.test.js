import { Input } from '../Input';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Input component', () => {
  test('it matches the snapshot', () => {
    const tree = renderer.create(<Input
      value='placeholder'
      autoComplete=''
      id='id'
      inputBoxLabel='>'
      key={123}
      label='MyInput'
      name="htmlName"
      onChange={() => {}}
      placeholder="placeholder"
      readOnly={false}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
