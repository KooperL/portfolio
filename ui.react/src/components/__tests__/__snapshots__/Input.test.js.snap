// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component it matches the snapshot 1`] = `
<div
  id="Input-container"
>
  <label
    htmlFor="htmlName"
  >
    MyInput
  </label>
  <div
    id="Input"
  >
    <div
      id="Input-decorator"
    >
      <p>
        &gt;
        Â 
      </p>
    </div>
    <input
      autoComplete=""
      id="id"
      name="htmlName"
      onChange={[Function]}
      placeholder="placeholder"
      readOnly={false}
      type="text"
      value="placeholder"
    />
  </div>
</div>
`;
